optimum/utils/preprocessing/text_classification.py

```diff
@@ -119,10 +119,9 @@ def run_inference(self, eval_dataset: Dataset, pipeline: TextClassificationPipel
             preds = pipeline.postprocess(model_outputs, function_to_apply=ClassificationFunction.NONE)
 
             if not self.task_args["is_regression"]:
-                # the dataset label ids may be different from the label2id of predictions
                 if self.label_to_id is not None:
-                    preds = self.config.label2id[preds["label"]]
-                    preds = self.label_to_id[preds]
+                    preds = int(self.config.label2id[preds["label"]])
+                    preds = self.label_to_id[preds]  # dataset label ids than of the model
```

Comment:
I think a word is missing in the comment.

Reply:
Thanks a lot! I changed the comment.